### ISSUE TITLE
load_module_async does not return an error if there are issues with loading or executing the module

### ISSUE THREAD

**Issue by utyfua** [rscarson/rustyscript#249]
**Describe the bug**

`load_module_async` does not return an error (`Error`) if there are issues with loading or executing the module [[ref](https://github.com/rscarson/rustyscript/blob/aa9113fb7223d285f917a733066d3d653108374f/src/runtime.rs#L804)]

`load_module` works as expected.

**To Reproduce**

1. Clone repo

2. Update example [examples/multiple_modules.rs]

```diff
diff --git a/examples/multiple_modules.rs b/examples/multiple_modules.rs
index db83bb4..f685be3 100644
--- a/examples/multiple_modules.rs
+++ b/examples/multiple_modules.rs
@@ -17,6 +17,8 @@ const API_MODULE: StaticModule = module!(
     export function setValue(value:number) {
       my_internal_value = value * 2;
     }
+
+    throw new Error('Test error');
   "
 );
 
@@ -39,7 +41,15 @@ fn main() -> Result<(), Error> {
     // made with less overhead.
     // Just like before, `::<Undefined` means we do not care if the
     // function returns a result.
-    let module_handle = runtime.load_module(&API_MODULE.to_module())?;
+    let module_handle = runtime.tokio_runtime().block_on(async {
+        runtime
+            .load_module_async(&API_MODULE.to_module())
+            .await
+            .expect("[1] EXPECTED TO FAIL HERE")
+    });
+
+    panic!("[2] SHOULD FAIL ABOVE THIS LINE");
+
     runtime.call_entrypoint::<Undefined>(&module_handle, json_args!(2))?;
 
     // Now we can load our new module from the filesystem

```

3. `cargo run --example multiple_modules`

**Actual output**
```
thread 'main' panicked at rustyscript\examples\multiple_modules.rs:51:5:
[2] SHOULD FAIL ABOVE THIS LINE
```

**Expected output**

If revert changes before #245 fix ([permalink](https://github.com/rscarson/rustyscript/tree/988c196040bda5d34ad5b56e794a4f369ba6bcde)) or use `load_module` then output will be
```
thread 'main' panicked at rustyscript\examples\multiple_modules.rs:48:12:
[1] EXPECTED TO FAIL HERE: JsError(JsError { name: Some("Error"), message: Some("Test error"), stack: Some("Error: Test error\n    at file:///<...>/rustyscript/examples/get_value.ts:12:9"), cause: None, exception_message: "Uncaught (in promise) Error: Test error", frames: [JsStackFrame { type_name: None, function_name: None, method_name: None, file_name: Some("file:///<...>/rustyscript/examples/get_value.ts"), line_number: Some(12), column_number: Some(9), eval_origin: None, is_top_level: Some(true), is_eval: false, is_native: false, is_constructor: false, is_async: false, is_promise_all: false, promise_index: None }], source_line: None, source_line_frame_index: Some(0), aggregated: None })
```

**Enviroment:**
 - OS: Windows 10.0.19045.4780
 - Tree without issue: [permalink](https://github.com/rscarson/rustyscript/tree/988c196040bda5d34ad5b56e794a4f369ba6bcde)
 - Tree with issue: [permalink](https://github.com/rscarson/rustyscript/tree/aa9113fb7223d285f917a733066d3d653108374f)

**Comment by rscarson** [comment]
Good catch! Was missing a `tokio::select` call in the side-module branch of the load_modules function

**Comment by rscarson** [comment]
Closed in v0.9.0
